src/report_1.R

```diff
@@ -1,3 +1,169 @@
+        group_by(Simulation) %>%
+        summarise(n = n())
+      quants <- quantile(counts_by_simulation$n, probs = c(0.05, 0.25, 0.5, 0.75, 0.975))
+      projected_totals <- rbind(projected_totals, data.frame(date = c(as.Date(date)), lower.ci = c(quants[1]), 
+                                                             q1 = c(quants[2]), median = c(quants[3]), 
+                                                             q3 = c(quants[4]), upper.ci = c(quants[5]),
+                                                             variable = c("Projected Median")))
+    }
+    projected_totals <- projected_totals %>%
+      mutate(date = ymd(date)) %>% 
+      filter(lower.ci != upper.ci)
+  }
+  
+  actual_totals <- data.frame(date = c(), variable = c(), value = c())
+  for (date in dates[dates < end_date]) {
+    counts <- actual_episodes %>%
+      filter(report_date <= date & (is.na(ceased) | ceased > date)) %>%
+      summarise(n = n())
+    actual_totals <- rbind(actual_totals, data.frame(date = c(as.Date(date)), variable = c("SSDA903"), value = c(counts[[1]])))
+  }
+
+  if(!is.null(counts_file)) {
+    counts <- process_counts(counts_file)
+    actual_totals <- bind_rows(actual_totals, counts %>% 
+                                 select(date, Total) %>% 
+                                 mutate(variable = "MIS") %>% 
+                                 rename(value = Total) %>% 
+                                 mutate(value = as.numeric(as.character(value))))
+  } 
+  
+  print(ggplot() +
+          {if(!is.null(projected_episodes_file)) 
+            list(
+              geom_ribbon(data = projected_totals %>% mutate(variable = "Projected IQR"), 
+                          aes(x = date, ymin = q1, ymax = q3, colour = variable), 
+                          linetype = 0, alpha = 0.4, show.legend = FALSE),
+              geom_ribbon(data = projected_totals %>% mutate(variable = "95% Simulation Range"), 
+                          aes(x = date, ymin = lower.ci, ymax = upper.ci, colour = variable), 
+                          linetype = 0, alpha = 0.2, show.legend = FALSE),
+              geom_line(data = projected_totals, 
+                        aes(x = date, y = median, colour = variable), 
+                        linetype = 2))} +
+          geom_line(data = actual_totals, aes(x = date, y = value, colour = variable)) +
+          geom_vline(xintercept = train_from, color = "black", linetype = 3, alpha = 0.5) +
+          geom_vline(xintercept = project_from, color = "black", linetype = 3, alpha = 0.5) +
+          theme_mastodon +
+          scale_color_manual(values = colours, 
+                             limits=c("SSDA903", "MIS", "Projected Median", "Projected IQR", "95% Simulation Range")) +
+          labs(title = "Total Children in Care", x = "Date", y = "No. children in care", colour = "Dataset", 
+               caption = " e.g. timestamp of the chart being generated? name of LA? date of model run? version of model?")
```

Comment:
I'd like to get to the point where this ggplot bit is the only bit we are doing in R for this. I think all the data manipulation can be done in clojure. It is the kind of thing I'd like to have in clojisr.

Is it worth having a separate data prep and charting function?

Reply:
Yes, in short. I'd was thinking when I get to doing the next round of charting it will become more clear what is generic and can be reused and functionalised. That will also making porting  data prep to clj easier